### PR TITLE
Add identity server test

### PR DESCRIPTION
**Pull request by erikjohnston**

### CODE REVIEW

File: tests/54identity.pl [hunk 1]
```diff
@@ -0,0 +1,31 @@
+test "Can bind 3PID via home server",
```
[the remark this answers is not in the document]
Comment (erikjohnston) [reply tests/54identity.pl]: > probably ought to have a copyright comment.

Almost nowhere else has copyright notices, which may or may not be problematic.

> Also: is it worth starting a new subdir?

I don't think so tbh, we can always do it down the line?

File: tests/30rooms/12thirdpartyinvite.pl [hunk 2]
```diff
@@ -12,7 +12,7 @@
 
       my $room_id;
 
-      $id_server->bind_identity( undef, "email", $invitee_email, $invitee )
+      $id_server->bind_identity( undef, "email", $invitee_email, $invitee->user_id )
```
[the remark this answers is not in the document]
Comment (erikjohnston) [reply tests/30rooms/12thirdpartyinvite.pl]: Oh cool, do you have an example? My greps are failing me :(

[the remark this answers is not in the document]
Comment (erikjohnston) [reply tests/30rooms/12thirdpartyinvite.pl]: Nevermind, found it!

File: lib/SyTest/Identity/Server.pm [hunk 3]
```diff
@@ -140,6 +140,31 @@ sub on_request
...
+      my $body = $req->body_from_form;
+      my $sid = $body->{sid};
+      my $mxid = $body->{mxid};
+
+      my $medium = $self->{validated}{$sid}{medium};
+      my $address = $self->{validated}{$sid}{address};
+
+      $self->bind_identity( undef, $medium, $address, $mxid );
+
+      $resp{medium} = $self->{validated}{$sid}{medium};
```
[the remark this answers is not in the document]
Comment (erikjohnston) [reply lib/SyTest/Identity/Server.pm]: Consistency ftw here I think

File: tests/54identity.pl (hunk 1 above)
[the remark this answers is not in the document]
Comment (richvdh) [reply tests/54identity.pl]: > Almost nowhere else has copyright notices, which may or may not be problematic.

well, they should do, and we should get into the habit of adding them on new files. Don't mind leaving it for now though

> > Also: is it worth starting a new subdir?
>
> I don't think so tbh, we can always do it down the line?

k then

**Pull request merged**